### PR TITLE
Bump trietools gomod to rc 1.4.0

### PR DESCRIPTION
**Pull request by BeniaminDrasovean**

### CODE REVIEW

File: trieTools/accountStorageExporter/main.go [hunk 1]
```diff
@@ -159,6 +163,11 @@ func exportStorage(address string, flags config.ContextFlagsConfigAddr, mainRoot
 		keyValueMap[hex.EncodeToString(leaf.Key())] = hex.EncodeToString(value)
 	}
 
+	err = common.GetErrorFromChanNonBlocking(iteratorChannels.ErrChan)
```
Comment (raduchis) [inline trieTools/accountStorageExporter/main.go:166]: :+1:

File: trieTools/go.mod [hunk 2]
```diff
@@ -3,47 +3,59 @@ module github.com/ElrondNetwork/elrond-tools-go/trieTools
...
 	golang.org/x/sys v0.0.0-20220412211240-33da011f77ad // indirect
+	golang.org/x/text v0.3.7 // indirect
 	google.golang.org/protobuf v1.28.0 // indirect
+	gopkg.in/yaml.v2 v2.4.0 // indirect
 	gopkg.in/yaml.v3 v3.0.1 // indirect
 )
 
 replace github.com/gogo/protobuf => github.com/ElrondNetwork/protobuf v1.3.2
-
-replace github.com/ElrondNetwork/arwen-wasm-vm/v1_2 v1.2.40 => github.com/ElrondNetwork/arwen-wasm-vm v1.2.40
```
Comment (raduchis) [inline trieTools/go.mod:45]: not needed any more?

**Review by raduchis — APPROVED** [review]

File: trieTools/trieToolsCommon/common.go [hunk 3]
```diff
@@ -131,20 +134,22 @@ func CreatePruningStorer(flags ContextFlagsConfig, maxDBValue int) (storage.Stor
...
-		DbPath:                    "",
-		PersisterFactory:          factory.NewPersisterFactory(localDbConfig),
-		Notifier:                  notifier.NewManualEpochStartNotifier(),
-		OldDataCleanerProvider:    &testscommon.OldDataCleanerProviderStub{},
-		CustomDatabaseRemover:     disabled.NewDisabledCustomDatabaseRemover(),
-		MaxBatchSize:              45000,
-		NumOfEpochsToKeep:         uint32(maxDBValue) + 1,
-		NumOfActivePersisters:     uint32(maxDBValue) + 1,
-		StartingEpoch:             uint32(maxDBValue),
+	args := pruning.StorerArgs{
```
Comment (raduchis) [inline trieTools/trieToolsCommon/common.go:137]: the NewPersistersTracker is missing; could be added like this:

`epochsData := pruning.EpochArgs{
		NumOfEpochsToKeep:     uint32(maxDBValue) + 1,
		NumOfActivePersisters: uint32(maxDBValue) + 1,
		StartingEpoch:         uint32(maxDBValue),
	}

	args := pruning.StorerArgs{
		Identifier:                "",
		ShardCoordinator:          testscommon.NewMultiShardsCoordinatorMock(1),
		CacheConf:                 cacheConfig,
		PathManager:               components.NewSimplePathManager(dbPath),
		DbPath:                    "",
		PersisterFactory:          factory.NewPersisterFactory(localDbConfig),
		Notifier:                  notifier.NewManualEpochStartNotifier(),
		OldDataCleanerProvider:    &testscommon.OldDataCleanerProviderStub{},
		CustomDatabaseRemover:     disabled.NewDisabledCustomDatabaseRemover(),
		MaxBatchSize:              45000,
		EpochsData:                epochsData,
		PruningEnabled:            true,
		EnabledDbLookupExtensions: false,
		PersistersTracker:         pruning.NewPersistersTracker(epochsData),
	}`

**Review by raduchis — CHANGES_REQUESTED** [review]

File: trieTools/trieToolsCommon/common.go [hunk 4]
```diff
@@ -174,15 +182,37 @@ func CreateTrie(storer storage.Storer) (common.Trie, error) {
...
-	tsm, err := trie.NewTrieStorageManagerWithoutPruning(storer)
+	tsm, err := CreateStorageManager(storer)
 	if err != nil {
 		return nil, err
 	}
 
 	return trie.NewTrie(tsm, Marshaller, Hasher, maxTrieLevelInMemory)
 }
 
+func CreateStorageManager(storer storage.Storer) (common.StorageManager, error) {
```
[the remark this answers is not in the document]
Comment (BeniaminDrasovean) [reply trieTools/trieToolsCommon/common.go:193]: Added.

File: trieTools/trieChecker/main.go [hunk 5]
```diff
@@ -108,8 +109,11 @@ func checkTrie(flags trieToolsCommon.ContextFlagsConfig, mainRootHash []byte) er
 		return err
 	}
 
-	ch := make(chan core.KeyValueHolder, common.TrieLeavesChannelDefaultCapacity)
-	err = tr.GetAllLeavesOnChannel(ch, context.Background(), mainRootHash)
+	iteratorChannels := &common.TrieIteratorChannels{
```
[the remark this answers is not in the document]
Comment (BeniaminDrasovean) [reply trieTools/trieChecker/main.go:112]: Done.

**Review by iulianpascalau — APPROVED** [review]

**Review by raduchis — APPROVED** [review]

**Pull request merged**